libs/service/api_service.cpp

```diff
@@ -64,3 +237,162 @@ ebpf_verify_program(
 
     return result;
 }
+
+ebpf_result_t
+ebpf_verify_and_load_program(
+    const GUID* program_type,
+    ebpf_handle_t program_handle,
+    ebpf_execution_context_t execution_context,
+    ebpf_execution_type_t execution_type,
+    uint32_t handle_map_count,
+    fd_handle_map* handle_map,
+    uint32_t byte_code_size,
+    uint8_t* byte_code,
+    const char** error_message,
+    uint32_t* error_message_size)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+    int error = 0;
+    uint64_t log_function_address;
+    // EbpfMapDescriptor descriptor;
+    struct ubpf_vm* vm = nullptr;
+    ebpf_protocol_buffer_t request_buffer;
+    ebpf_operation_load_code_request_t* request = nullptr;
+
+    // Only kernel execution context supported currently.
+    if (execution_context == execution_context_user_mode) {
+        return EBPF_INVALID_ARGUMENT;
+    }
+
+    *error_message = nullptr;
+    *error_message_size = 0;
+
+    clear_map_descriptors();
+
+    // Query map descriptors from execution context.
+    try {
+        result = _query_and_cache_map_descriptors(handle_map, handle_map_count);
+        if (result != EBPF_SUCCESS) {
+            goto Exit;
+        }
+
+        // Verify the program
+        error = verify_byte_code(program_type, byte_code, byte_code_size, error_message, error_message_size);
+
+        if (error != 0) {
+            result = EBPF_VALIDATION_FAILED;
+            goto Exit;
+        }
+
+        if (byte_code_size > MAX_CODE_SIZE) {
+            result = EBPF_PROGRAM_TOO_LARGE;
+            goto Exit;
+        }
+
+        ebpf_code_buffer_t byte_code_buffer(byte_code, byte_code + byte_code_size);
+
+        result = _resolve_maps_in_byte_code(program_handle, byte_code_buffer);
+        if (error != ERROR_SUCCESS) {
+            goto Exit;
+        }
+
+        result = _resolve_ec_function(EBPF_EC_FUNCTION_LOG, &log_function_address);
+        if (result != EBPF_SUCCESS) {
+            goto Exit;
+        }
+
+        if (execution_type == EBPF_EXECUTION_JIT) {
+            std::map<uint32_t, uint64_t> helper_id_to_adddress;
+            result = _build_helper_id_to_address_map(program_handle, byte_code_buffer, helper_id_to_adddress);
+            if (result != EBPF_SUCCESS)
+                goto Exit;
+
+            ebpf_code_buffer_t machine_code(MAX_CODE_SIZE);
+            size_t machine_code_size = machine_code.size();
+
+            // JIT code.
+            vm = ubpf_create();
+            if (vm == nullptr) {
+                result = EBPF_JIT_COMPILATION_FAILED;
+                goto Exit;
+            }
+
+            for (const auto& helper : helper_id_to_adddress) {
+                if (ubpf_register(vm, helper.first, nullptr, reinterpret_cast<void*>(helper.second)) < 0) {
```

Comment:
see earlier comments about .first and .second.  And look for anywhere else in this PR that uses them.